ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -197,18 +202,26 @@ public BlobStore(ReplicaId replicaId, StoreConfig config, ScheduledExecutorServi
     this.accountService = accountService;
     this.replicaStatusDelegates = config.storeReplicaStatusDelegateEnable ? replicaStatusDelegates : null;
     this.time = time;
-    long threshold = config.storeReadOnlyEnableSizeThresholdPercentage;
-    long delta = config.storeReadOnlyToPartialWriteEnableSizeThresholdPercentageDelta;
-    this.thresholdBytesHigh = (long) (capacityInBytes * (threshold / 100.0));
-    this.thresholdBytesLow = (long) (capacityInBytes * ((threshold - delta) / 100.0));
+    this.sealThresholdBytesHigh =
+        (long) (capacityInBytes * (config.storeReadOnlyEnableSizeThresholdPercentage / 100.0));
+    this.sealThresholdBytesLow = (long) (capacityInBytes * (
+        (config.storeReadOnlyEnableSizeThresholdPercentage
+            - config.storeReadOnlyToPartialWriteEnableSizeThresholdPercentageDelta) / 100.0));
+    this.partialSealThresholdBytesHigh =
+        (long) (capacityInBytes * (config.storePartialWriteEnableSizeThresholdPercentage / 100.0));
+    this.partialSealThresholdBytesLow = (long) (capacityInBytes * (
+        (config.storePartialWriteEnableSizeThresholdPercentage
+            - config.storePartialWriteToReadWriteEnableSizeThresholdPercentageDelta) / 100.0));
```

Comment:
Currently we haven't fully supported the partially seal yet.
After we merge this code, how do we make sure there is no impact to the existing code logic?

Reply:
What's the default value of partialSealThresholdBytesLow and storePartialWriteToReadWriteEnableSizeThresholdPercentageDelta we'll set?

Reply:
The default value for `storePartialWriteEnableSizeThresholdPercentage` is 50%. And the default value for `storePartialWriteToReadWriteEnableSizeThresholdPercentageDelta` is 5%. They are defined in StoreConfig.java.

Reply:
> After we merge this code, how do we make sure there is no impact to the existing code logic?

If you check the `ReplicaStatusDelegate::partialSeal()` implementation, it does the same thing that `ReplicaStatusDelegate::unseal()` does. So effectively, anything that is not sealed will remain not sealed.

Reply:
I think after we set the partialSeal(), even the implementation is to set as unseal, the metrics will start to show as partial sealed.

I'm wondering if we set storePartialWriteEnableSizeThresholdPercentage to storeReadOnlyEnableSizeThresholdPercentage, looks like we won't generate partial seal status at all.

I leave it to you.